### PR TITLE
[CodeQuality] Add InlineArrayReturnAssignRector

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: src/Rector/AbstractRector.php [hunk 1]
```diff
@@ -132,7 +129,6 @@ public function autowire(
         PhpVersionProvider $phpVersionProvider,
         ExclusionManager $exclusionManager,
         StaticTypeMapper $staticTypeMapper,
-        ParameterProvider $parameterProvider,
```
Comment (samsonasik) [inline src/Rector/AbstractRector.php]: it need to be kept to avoid issue https://github.com/rectorphp/rector/issues/7117 happen again.

Comment (TomasVotruba) [reply src/Rector/AbstractRector.php]: I see. We should come up with a unprefix or better scoped method then.

Comment (TomasVotruba) [reply src/Rector/AbstractRector.php]: How exactly do you use the parameter provider?

Comment (samsonasik) [reply src/Rector/AbstractRector.php]: it used in laminas-servicemanager-migration to replace use interop statement when auto import enabled: https://github.com/laminas/laminas-servicemanager-migration/blob/93cfd29ea5cf09fc634db9bf63f98ab377da0560/src/Rector/Class_/ImplementsFactoryInterfaceToPsrFactoryRector.php#L161-L188

**Pull request merged**